sdk/rust/oak/src/io/receiver.rs

```diff
@@ -66,14 +66,33 @@ impl<T: Decodable> Receiver<T> {
     }
 
     /// Wait for a value to be available from this receiver.
+    ///
+    /// Returns [`ChannelReadStatus`] of the wrapped handle, or `Err(OakStatus::ErrTerminated)` if
+    /// the Oak Runtime is terminating,
     #[allow(clippy::trivially_copy_pass_by_ref)]
-    pub fn wait(&self) -> Result<(), OakStatus> {
+    pub fn wait(&self) -> Result<ChannelReadStatus, OakStatus> {
         // TODO(#500): Consider creating the handle notification space once and for all in `new`.
         let read_handles = vec![self.handle];
         let mut space = crate::new_handle_space(&read_handles);
         crate::prep_handle_space(&mut space);
         let status =
             unsafe { oak_abi::wait_on_channels(space.as_mut_ptr(), read_handles.len() as u32) };
-        crate::result_from_status(status as i32, ())
+
+        match crate::result_from_status(status as i32, ()) {
+            Err(OakStatus::ErrTerminated) => Err(OakStatus::ErrTerminated),
+            Err(OakStatus::ErrInvalidArgs) => panic!("Indicates that `space` is corrupted. This should never happen."),
+            // The only other output from `oak_abi::wait_on_channels` is `OakStatus::Ok`, 
+            // which is handled below. Other `OakStatus` error codes never happen.
```

Comment:
Is it possible to add the logic that shows this is actually the case?

```rust
Err(_) => panic!("should never happen")
Ok(...) => space...
```

Reply:
I tried that, but `clippy` does not like it:
```
error: `Err(_)` matches all errors
  --> sdk/rust/oak/src/io/receiver.rs:84:13
   |
84 |             Err(_) => panic!("Other errors never happen."),
   |             ^^^^^^
   |
   = note: `-D clippy::match-wild-err-arm` implied by `-D warnings`
   = note: match each error separately or use the error output
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_wild_err_arm
```
I can still add a separate arm for the `Ok` case.

Reply:
Yeah that's even better!